### PR TITLE
Document CA, CRLs bundling and client auth setup for nginx and haproxy

### PR DESCRIPTION
**Pull request by gwarf**
Document CA, CRLs bundling and client auth setup for nginx and haproxy.

Client authentication is required for Nginx and HAproxy to make them present the supported CAs' DNs to the [IGTF Monitoring probe](https://github.com/ARGOeu/nagios-plugins-igtf/blob/master/src/check_igtf_ca) that is still enable until OIDC will be the only authentication mean.

### CODE REVIEW

**Comment by gwarf** [comment]
I wonder if Client authentication isn't also required for Apache HTTPd to present the CAs DNs...

File: source/openstack.rst [hunk 1]
```diff
@@ -175,17 +175,62 @@ Pre-requisites
...
+     HTTPd is able to use CAs and CRLs contained in a directory
+     ::
 
       SSLCACertificatePath    /etc/grid-security/certificates
       SSLCARevocationPath     /etc/grid-security/certificates
 
-   IGTF CAs can be obtained from UMD, you can find repo files for your distribution
-   at http://repository.egi.eu/sw/production/cas/1/current/
-
+   For Nginx
```
Comment (enolfc) [inline source/openstack.rst]: I think there is no easy way to run Keystone + oidc on NGINX, should we really include it?

Comment (gwarf) [reply source/openstack.rst]: OK, I was understanding that some sites may have been using it, I agree that if it's not the case we can drop this and keep only haproxy doc.

**Comment by gwarf** [comment]
For the split PR I would prefer avoiding doing it now, lacking time to rewrite history and so on... 
But I promise to take of this next time :)

File: source/openstack.rst (hunk 1 above)
Comment (gwarf) [reply source/openstack.rst]: Done.

**Comment by gwarf** [comment]
OK, so extracted change related to building doc to #29.

**Comment by gwarf** [comment]
For nginx + oidc, couldn't we use https://github.com/zmartzone/lua-resty-openidc ?
Looks like it was made by the same person doing https://github.com/zmartzone/mod_auth_openidc
https://hanszandbelt.wordpress.com/2015/08/14/openid-connect-for-nginx/

**Comment by enolfc** [comment]
> For nginx + oidc, couldn't we use https://github.com/zmartzone/lua-resty-openidc ?

Probably yes, but no one on fedcloud has tested it. I would not add it until we do some actual tests.

**Pull request merged**